### PR TITLE
Added tunnel options to vpn_connection

### LINKED ISSUE #2220: Feature request: aws_vpn_connection add pre shared keys to attributes
_This issue was originally opened by @wolfgang-werner as hashicorp/terraform#16595. It was migrated here as a result of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

AWS recently added the possibility to submit predefined preshared keys and inside IP for both VPN tunnels. 

Please add these attributes to the aws_vpn_connection.

With this attributes it is possible to create and destroy the vpn connection without the need to recreate the vpn tunnel on the customer gateway. The AWS behaviour without these attributes is to generate new keys everytime.

Thanks, Wolfgang

### PR DESCRIPTION
**Pull request by a-teisseire**
Closes #2220

### Context

AWS recently allowed users to let them choose the CIDR and the PSK to use inside IPsec tunnels.
More details here : https://aws.amazon.com/about-aws/whats-new/2017/10/aws-vpn-update-custom-psk-inside-tunnel-ip-and-sdk-update/

### Changes

This Pull Request changes the way tunnel1_preshared_key and tunnel2_preshared_key behave (from read-only to writeable). 
The tunnel1_inside_cidr and tunnel2_inside_cidr have been introduced also to be used as parameters for the CreateVpnConnection API.
Test coverage has been added to test these features.

### Tests

```
⇒  TF_ACC=1 go test ./aws -v -run=TestAccAWSVpnConnection_ -timeout 120m 
=== RUN   TestAccAWSVpnConnection_importBasic 
--- PASS: TestAccAWSVpnConnection_importBasic (236.43s) 
=== RUN   TestAccAWSVpnConnection_basic 
--- PASS: TestAccAWSVpnConnection_basic (432.65s) 
=== RUN   TestAccAWSVpnConnection_tunnelOptions 
--- PASS: TestAccAWSVpnConnection_tunnelOptions (286.92s) 
=== RUN   TestAccAWSVpnConnection_withoutStaticRoutes 
--- PASS: TestAccAWSVpnConnection_withoutStaticRoutes (298.02s) 
=== RUN   TestAccAWSVpnConnection_disappears 
--- PASS: TestAccAWSVpnConnection_disappears (166.75s) 
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	1420.829s
```

### CODE REVIEW

**Comment by tyrken** [comment]
FYI I compiled this locally and it Worked-For-Me&trade; when I needed it.

**Comment by bflad** [comment]
This PR would close #2220

File: aws/resource_aws_vpn_connection.go [hunk 1]
```diff
@@ -245,8 +261,37 @@ func resourceAwsVpnConnection() *schema.Resource {
 func resourceAwsVpnConnectionCreate(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).ec2conn
 
+	// Get the optional tunnel options
+	tunnel1_cidr := d.Get("tunnel1_inside_cidr").(string)
```
Comment (bflad) [inline aws/resource_aws_vpn_connection.go]: These can be simplified by using `d.GetOk()` and placed where you have your other if statements 😄 

e.g.
```
if v, ok := d.GetOk("tunnel1_preshared_key"); ok {
    options[0].TunnelInsideCidr = aws.String(v.(string))
}
```

File: aws/resource_aws_vpn_connection_test.go [hunk 2]
```diff
@@ -325,6 +359,38 @@ func testAccAwsVpnConnectionConfigUpdate(rInt, rBgpAsn int) string {
 	`, rBgpAsn, rInt)
 }
 
+func testAccAwsVpnConnectionConfigTunnelOptions(rBgpAsn int) string {
+	return fmt.Sprintf(`
+		resource "aws_vpn_gateway" "vpn_gateway" {
```
Comment (bflad) [inline aws/resource_aws_vpn_connection_test.go]: Nitpick: I don't think we have a style guide I can point to, but can you please ensure the Terraform configuration inside the testing here is fully to the left with 2 space indentation? We're trying to standardize going forward. Thanks!

**Comment by a-teisseire** [comment]
Hey,

Thanks a lot for this feedback, I'll try to commit an update with these details today or in the upcoming days.

**Comment by radeksimko** [comment]
Hi @a-teisseire 
thanks for the PR!

Do you need any help implementing the changes @bflad mentioned?

**Comment by a-teisseire** [comment]
Hello @radeksimko 

I have implemented the changes and made the documentation.
@bflad changes were no difficulty to implement. I re ran the tests on my side, everything should be good now !

**Comment by bflad** [comment]
This has been released in [terraform-provider-aws version 1.8.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#180-january-29-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**